### ISSUE TITLE
Hosting upload fails with timeout

### ISSUE THREAD

**Issue by LanderBeeuwsaert** [firebase/firebase-tools#5449]
**firebase-tools:**
11.16.1
windows 10

PS C:\xyz> firebase deploy -P e2 --only hosting --public deploy_e2_dist

=== Deploying to 'some-project'...

i  deploying hosting
i  hosting[some-project]: beginning deploy...
i  hosting[some-project]: found 6022 files in deploy_e2_dist
-  hosting: uploading new files [2/19] (10%)
Error: Task 0f3e601b21cbbaeec4fb1b151a302c088b3d2f366fb2c0bc8cae267a01ceb552 failed: retries exhausted after 6 attempts,
 with error: Timeout reached making request to https://upload-firebasehosting.googleapis.com/upload/sites/some-project/versions/4a45336e082721f5/files/0f3e601b21cbbaeec4fb1b151a302c088b3d2f366fb2c0bc8cae267a01ceb552
|  hosting: uploading new files [2/19] (10%)

Hi, we have quite a large project (6022 files and counting) and when uploading to firebase hosting we get timeouts.
After redoing the upload multiple times, each time it can upload a few files and eventually (+-10x) it succeeds.
However when we tried to put a big video file in the project, that video file would fail to upload due to timeout even.
Would be great that each time a file is successfully uploaded the timeout timer is reset? + that you can configure the timeout timer?

**Comment by bkendall** [comment]
Can you give me a little more information?

1. How big of a video file are you uploading? Note, since Hosting charges for bandwidth, video files can get expensive pretty quickly to host. Usually I recommend serving the video some other way (embed YouTube, reference from Cloud Storage, etc). 
2. IIRC, the timeout is per-file, so it's likely the single file is taking too long to upload. It looks like the timeout for an individual file is based on the size of the file (20s per MB). There's no way to override this at the moment, so I'd be curious to know if your upload bandwidth has the capacity to accomplish this upload? Even if the connection is somewhat unstable, it may just fail. How long does the upload take before it does time out?

**Comment by LanderBeeuwsaert** [comment]
Hi, I just did a search and sorted on size, so 30MB is the biggest one and it goes down quickly.
![image](https://user-images.githubusercontent.com/1098243/214404725-2c1b8877-534e-43bc-ba00-1c928878c202.png)
Now that I see it, we indeed don't have the video files anymore in the project (that was 1 or 2 years ago) so our biggest files are some pdfs and pictures.

I have good internet (results are using WIFI, using cable it would probably be better but because I never have issues I don't even plug it in)
![image](https://user-images.githubusercontent.com/1098243/214405504-4be7a89e-24d4-4c9b-94f2-eaf3c7021838.png)

So something feels of IMO, could it be related to the fact that we have so many files that something in the indexing goes haywire? (we have thousands of small (5 => 40kb) image files)
